### PR TITLE
docs: add note in upgrade guide about cephfs driver's breaking change

### PR DESCRIPTION
**Pull request by Rakshith-R**
**Description of your changes:**

CephCSI CephFS introduced a breaking change in v3.9.0. This commit adds a note in upgrade guide regarding the same.

refer: https://github.com/ceph/ceph-csi/blob/v3.9.0/docs/ceph-csi-upgrade.md/#upgrading-cephfs

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: Documentation/Upgrade/rook-upgrade.md [hunk 1]
```diff
@@ -49,6 +49,9 @@ those releases.
 ## Breaking changes in v1.12
 
 * The minimum supported version of Kubernetes is v1.22.
+* CephCSI CephFS driver introduced a breaking change in v3.9.0. Please follow the steps mentioned in the
```
Comment (travisn) [inline Documentation/Upgrade/rook-upgrade.md:52]: Can you summarize the breaking change in one or two sentences here? In this Rook upgrade guide, they should understand if they are affected, and if it's common. By default, users won't be affected, right? I don't expect it's common to use the mount options. For example, say that if they are setting the mountOptions in a CephFS storage class, they need to follow those instructions in the CSI upgrade guide.

**Review by travisn — CHANGES_REQUESTED** [review]

File: Documentation/Upgrade/rook-upgrade.md (hunk 1 above)
Comment (Rakshith-R) [reply Documentation/Upgrade/rook-upgrade.md]: :+1: yes
Updated mentioning the above suggested details, 
Ptal

**Review by travisn — APPROVED** [review]
